src/Arrays/MPIStateArrays.jl

```diff
@@ -20,6 +20,13 @@ Base.similar(::Type{A}, ::Type{FT}, dims...) where {A <: Array, FT} =
     Base.similar(::Type{A}, ::Type{FT}, dims...) where {A <: CuArray, FT} =
         similar(CuArray{FT}, dims...)
 end
+@init @require CUDAnative = "be33ccc6-a3ff-5ff2-a52e-74243cff1e17" begin
+    using .CUDAnative
+    using Adapt
+    # allows passing MPIStateArrays to GPU kernels
+    Adapt.adapt_storage(to::CUDAnative.Adaptor, a::MPIStateArray) =
+        adapt(to, a.data)
```

Comment:
Should this really be the realdata or all the data?
```suggestion
        adapt(to, a.realdata)
```

Reply:
I think that in general kernels may need access to all the data, e.g. `facerhs!` ?

Reply:
OK. Just wanted to check.